packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -150,6 +150,26 @@ export interface BasicCrawlerOptions<
      */
     handleRequestTimeoutSecs?: number;
 
+    /**
+     * A function to handle requests that failed less than `option.maxRequestRetries` times
+     * and would be retried by the crawler by default.
+     * Note that if the function is user-provided, it would not reclaim the request automatically.
+     *
+     * The function receives the following object as an argument:
+     * ```
+     * {
+     *     request: Request,
+     *     error: Error,
+     *     session: Session,
+     *     crawler: BasicCrawler,
+     * }
+     * ```
+     * where the {@link Request} instance corresponds to the request to be reclaimed, and the `Error` instance
+     * represents the last error thrown during processing of the request.
+     *
+     */
+    preReclaimRequestHandler?: FailedRequestHandler<ErrorContext>;
```

Comment:
The name is not very intuitive. I'd just go with `failedRequestHandler`. Request reclaiming is an implementation detail of the crawler.

Reply:
Or maybe just `errorHandler`, that's even more descriptive.